hoprd/rest-api/src/preconditions.rs

```diff
@@ -36,14 +37,18 @@ pub(crate) async fn authenticate(
                 .collect::<Vec<_>>();
 
             // We have multiple websocket routes that need authentication checks
-            let is_ws_auth =
-                if uri.path().starts_with("/messages/websocket") || uri.path().starts_with("/session/websocket") {
-                    uri.query()
-                        .unwrap_or("")
-                        .contains(&format!("apiToken={}", expected_token))
-                } else {
-                    false
-                };
+            let is_ws_auth = if uri.path().starts_with("/api/v3/messages/websocket")
+                || uri.path().starts_with("/api/v3/session/websocket")
+            {
+                uri.query()
+                    .map(|q| match decode(q) {
+                        Ok(decoded) => decoded.into_owned().contains(&format!("apiToken={}", expected_token)),
+                        Err(_) => false,
+                    })
+                    .unwrap_or(false)
```

Comment:
How does your solution help with timing based attacks? You even drop out early if decoded and expected token do not compare in length.

@Teebor-Choka I would expect the API framework to mitigate against long query strings aiming to DoS the service. Should this really be part of the authentication logic check?

Reply:
There must be some mechanism, since JS cannot do this in headers. It is not far fetched, because the websocket endpoint essentially is without any auth protection otherwise.